clang/include/clang/3C/ConstraintsGraph.h

```diff
@@ -173,14 +173,20 @@ class DataGraph
     return getNeighbors(D, DataSet, false);
   }
 
+  NodeType *findNode(Data D) {
+    if (NodeSet.find(D) != NodeSet.end())
+      return NodeSet[D];
```

Comment:
What happens if you index a set with a `D` that it's not in the set? Will that just give you `nullptr` ?